docs/topics/OAuth2.md

```diff
@@ -56,68 +41,229 @@ Scopes provide access to certain resources of a user's account. Your API client
+    "scope": "identify connections"
+}
+```
+
+## Bots
+
+So, what are bot accounts?
+
+### Bot vs User Accounts
+
+Discord's API provides a seperate type of user account dedicated to automation, called a bot account. Bot accounts can be created through the [applications page](#MY_APPLICATIONS/top), and are authenticated using a token (rather than a username and password). Unlike the normal OAuth2 flow, bot accounts have full access to all API routes without using bearer tokens, and can connect to the [Real Time Gateway](#DOCS_GATEWAY/gateways). Automating normal user accounts (generally called "self-bots") outside of the OAuth2/bot API is forbidden, and can result in an account termination if found.
+
+Bot accounts have a few differences in comparison to normal user accounts, namely:
+
+1. Bots are added to servers through the OAuth2 API, and cannot accept normal invites.
+2. Bots cannot have friends, nor be added to or join Group DMs.
+3. Bots do not have a maximum number of Guilds (unlike user accounts, which are limited to 100).
+4. Bots have an entirely separate set of [Rate Limits](#DOCS_RATE_LIMITS/rate-limits).
+
+### Bot Auth Flow
+
+Bot auth is a special server-less and callback-less OAuth2 flow that makes it easy for users to add bots to servers. The URL you create looks similar to what we use for full stack implementation:
+
+```
+https://discordapp.com/oauth2/authorize?client_id=157730590492196864&scope=bot&permissions=1
 ```
 
-`client_id` is your application's ID and `redirect_uri` is one of your application's URL-encoded redirect URIs.
+In the case of bots, the `scope` parameter should be set to `bot`. There's also a new parameter, `permissions`, which is an integer corresponding to the [permission calculations](#DOCS_PERMISSIONS/permissions-bitwise-permission-flags) for the bot. You'll also notice the absence of `response_type` and `redirect_uri`. Bot auth does not require these parameters because there is no need to retrieve the user's access token.
+
+When the user navigates to this page, they'll be prompted to add the bot to a server in which they have proper permissions. On acceptance, the bot will be added. Super easy!
+
+### Advanced Bot Auth
 
-When a user is directed to this URL, they are prompted to select a channel for the webhook to be placed in. Your application will receive an authorization code back in the querystring (as usual with the authorization code grant).
+Enterprising devs can add some complexity to bot auth. You can request additional scopes outside of "bot", which will prompt a continuation into a complete flow and add the ability to request the user's access token. If you request any scopes outside of `bot`, `response_type` is again mandatory; we will also automatically redirect the user to the first uri in your application's registered list unless `redirect_uri` is specified. A full OAuth2 flow is also required if you check "Require OAuth2 Code Grant" in your application's settings.
```

Comment:
drop `enterprising` .. this is also not adding complexity, but extending functionality/purpose

Reply:
use `bot` over "bot"

Reply:
`A full OAuth2 flow is also required if you check`

to 

`A full OAuth2 authorization code grant flow can also be required`

i would also prefer separating this small line to a separate paragraph, and explaining why it exists, what it solves.

Reply:
please also explain private/public bots!